concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs

```diff
@@ -630,185 +660,51 @@ instance (BlobStorable m a, BlobStorable m b, MHashableTo m H.Hash a) => BlobSto
     (r, v') <- storeUpdate v
     return (r, Some v')
 
--- | Blobbed is a fixed point of the functor `f` wrapped in references of type @ref@
-newtype Blobbed ref f = Blobbed {unblobbed :: ref (f (Blobbed ref f))}
-
--- Serialize instances, just wrap the Serialize instances of the underlying reference
-deriving instance (forall a. Serialize (ref a)) => Serialize (Blobbed ref f)
-
--- If a monad can manage references of type @ref@ then it can store values of type
--- @Blobbed ref f@ (just by serializing the inner references) into references of type
--- @ref@
-instance (MonadBlobStore m, forall a. Serialize (ref a)) => BlobStorable m (Blobbed ref f)
-
--- If a monad can store references of type @ref@ and a reference is serializable and nullable,
--- then it can store values of type @Nullable (Blobbed ref f)@ into references of type @ref@
-instance (MonadBlobStore m, forall a. HasNull (ref a), forall a. Serialize (ref a)) => BlobStorable m (Nullable (Blobbed ref f))
-
-type instance Base (Blobbed ref f) = f
-
-instance (Monad m, BlobStorable m (f (Blobbed BlobRef f))) => MRecursive m (Blobbed BlobRef f) where
-    -- Projecting the blobbed reference boils down to load the value it contains
-    mproject (Blobbed r) = loadRef r
-
-instance (Monad m, BlobStorable m (f (Blobbed BlobRef f))) => MCorecursive m (Blobbed BlobRef f) where
-    -- Embedding a reference into a Blobbed ref boils down to storing the reference
-    membed r = Blobbed <$> storeRef r
-
--- | A type that is an instance of @HasNull@ has a distinguished value that is considered a Null value.
-class HasNull ref where
-    refNull :: ref
-    isNull :: ref -> Bool
-
-instance HasNull (BlobRef a) where
-    refNull = BlobRef maxBound
-    isNull = (== refNull)
+-- |'BufferedFix' is a fixed-point combinator that uses a 'BufferedRef'.
```

Comment:
This comment is as cryptic as it used to be. It would be good to expand on the intended uses instead of the generic comment.